ocs_ci/framework/main.py

```diff
@@ -108,27 +153,106 @@ def init_ocsci_conf(arguments=None):
+    framework.config.reset_ctx()
+    for i in range(nclusters):
+        framework.config.switch_ctx(i)
+        process_ocsci_conf(common_argv + multicluster_conf[i][1:])
+        for j in range(len(multicluster_conf[i][1:])):
+            if multicluster_conf[i][j + 1].startswith("--"):
+                multicluster_conf[i][j + 1] = f"{multicluster_conf[i][j+1]}{i + 1}"
+        framework.config.multicluster_args.append(multicluster_conf[i][1:])
+        check_config_requirements()
+
+
+def tokenize_per_cluster_args(args, nclusters):
+    """
+    Seperate per cluster arguments so that parsing becomes easy
+
+    Params:
+        args: Combined arguments
+        nclusters(int): total number of clusters
+
+    Returns:
+        list of lists: Each cluster conf per list
+            ex: [[cluster1_conf], [cluster2_conf]...]
+
+    """
+    per_cluster_argv = list()
+    multi_cluster_argv = list()
+    common_argv = list()
+    cluster_ctx = False
+    regexp = re.compile(r"--cluster[0-9]+")
+    index = 0
+
+    for i in range(1, nclusters + 1):
+        while index < len(args):
+            if args[index] == f"--cluster{i}":
+                cluster_ctx = True
+            elif regexp.search(args[index]):
+                cluster_ctx = False
+                break
+            if cluster_ctx:
+                per_cluster_argv.append(args[index])
+            else:
+                common_argv.append(args[index])
+            index = index + 1
+        multi_cluster_argv.append(per_cluster_argv)
+        per_cluster_argv = []
+    return multi_cluster_argv, common_argv
 
 
 def main(argv=None):
     arguments = argv or sys.argv[1:]
     init_ocsci_conf(arguments)
-    pytest_logs_dir = utils.ocsci_log_path()
-    utils.create_directory_path(framework.config.RUN["log_dir"])
-    launch_name = reporting.get_rp_launch_name()
-    arguments.extend(
-        [
-            "-p",
-            "ocs_ci.framework.pytest_customization.ocscilib",
-            "-p",
-            "ocs_ci.framework.pytest_customization.marks",
-            "-p",
-            "ocs_ci.framework.pytest_customization.reports",
-            "--logger-logsdir",
-            pytest_logs_dir,
-            "--rp-launch",
-            launch_name,
-        ]
-    )
+    for i in range(framework.config.nclusters):
+        framework.config.switch_ctx(i)
+        pytest_logs_dir = utils.ocsci_log_path()
+        utils.create_directory_path(framework.config.RUN["log_dir"])
+        launch_name = reporting.get_rp_launch_name()
+        arguments.extend(
+            [
+                "-p",
+                "ocs_ci.framework.pytest_customization.ocscilib",
+                "-p",
+                "ocs_ci.framework.pytest_customization.marks",
+                "-p",
+                "ocs_ci.framework.pytest_customization.reports",
+                "--logger-logsdir",
+                pytest_logs_dir,
+                "--rp-launch",
+                launch_name,
+            ]
```

Comment:
Do we really want to do this per cluster? These seem like common arguments to me.